### ISSUE TITLE
Run-AlValidation with "preprocessorSymbols"

### ISSUE THREAD

**Issue by GeekEding** [microsoft/navcontainerhelper#1704]
**Describe the issue**
I think the Run-AlValidation from the actual BcContainerHelper can't use "preprocessorSymbols" from the app.json. There is an issue compiling the app. The compiler shows an error with an unknown field. This field ist inserted with Preprocessor:
![grafik](https://user-images.githubusercontent.com/40355993/107965735-270a2880-6fab-11eb-9fab-e30ea21daad6.png)

I can build and publish the package from VS Code without any errors. 

If I delete this preprocessor and use the code without, the compiler shows an error on the next table with integrated preprocessor so I think this will be the problem. 

![grafik](https://user-images.githubusercontent.com/40355993/107965509-e90d0480-6faa-11eb-9068-58564b5fb7f8.png)

**Scripts used to create container and cause the issue**
```
Run-AlValidation `
    -licenseFile "$license" `
    -previousApps @( "$prevapp" ) `
    -apps @( "$app" ) `
    -validateCurrent `
    -countries "de" `
    -affixes "OPP" `
    -supportedCountries "de" `
```

**Full output of scripts**
```
  _____                   _                                    _____                           _____            
 |  __ \                 (_)                 /\               / ____|                         / ____|           
 | |__) |   _ _ __  _ __  _ _ __   __ _     /  \   _ __  _ __| (___   ___  _   _ _ __ ___ ___| |     ___  _ __  
 |  _  / | | | '_ \| '_ \| | '_ \ / _` |   / /\ \ | '_ \| '_ \\___ \ / _ \| | | | '__/ __/ _ \ |    / _ \| '_ \ 
 | | \ \ |_| | | | | | | | | | | | (_| |  / ____ \| |_) | |_) |___) | (_) | |_| | | | (__  __/ |____ (_) | |_) |
 |_|  \_\__,_|_| |_|_| |_|_|_| |_|\__, | /_/    \_\ .__/| .__/_____/ \___/ \__,_|_|  \___\___|\_____\___/| .__/ 
                                   __/ |          | |   | |                                              | |    
                                  |___/           |_|   |_|                                              |_|    
Copying previous apps to packages folder

...

Using Symbols Folder: C:\ProgramData\BcContainerHelper\e5a23a23-9d36-4e2c-9868-fd53f50e33ce
Copying System.app from Container
Copying Microsoft_Application_17.4.21491.22091.app from Container
Copying Microsoft_Base Application_17.4.21491.22091.app from Container
Copying Microsoft_System Application_17.4.21491.22091.app from Container
Compiling...
.\alc.exe /project:"C:\ProgramData\BcContainerHelper\3d9dde47-5e49-49b7-86e3-e7cc42e55238" /packagecachepath:"C:\ProgramData\BcContainerHelper\e5a23a23-9d36-4e2c-9868-fd53f50e33ce" /out:"C:\P
rogramData\BcContainerHelper\3d9dde47-5e49-49b7-86e3-e7cc42e55238\gbedv GmbH & Co. KG_OPplus 365_17.1700.1.1.app" /analyzer:C:\build\vsix\extension\bin\Analyzers\Microsoft.Dynamics.Nav.AppSou
rceCop.dll /ruleset:C:\ProgramData\BcContainerHelper\3d9dde47-5e49-49b7-86e3-e7cc42e55238\ruleset.json /assemblyprobingpaths:"C:\Program Files (x86)\Microsoft Dynamics NAV\170\RoleTailored Cl
ient","C:\Program Files\Microsoft Dynamics NAV\170\Service","C:\Program Files (x86)\Open XML SDK\V2.5\lib","c:\Windows\Microsoft.NET\Assembly","C:\Test Assemblies\Mock Assemblies"
Microsoft (R) AL Compiler version 6.4.6.9919
Copyright (C) Microsoft Corporation. All rights reserved

Compilation started for project 'OPplus 365' containing '765' files at '16:6:18.509'.
C:\ProgramData\BcContainerHelper\9549347e-c792-4b85-9db2-6577627d46e9\src\src\Table%205157812%20-%20OPP%20Accounting%20Hint%20Text.al(65,19): error AL0186: Reference 'Type' in application obj
ect 'OPP Accounting Hint Text' does not exist

Compilation ended at '16:37:7.454'.

Unexpected error while validating app. Error is: App generation failed
bei Compile-AppInBcContainer, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\2.0.5\AppHandling\Compile-AppInNavContainer.ps1: Zeile 515
bei <ScriptBlock>, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\2.0.5\AppHandling\Run-AlCops.ps1: Zeile 68
bei <ScriptBlock>, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\2.0.5\AppHandling\Run-AlCops.ps1: Zeile 212
bei Run-AlCops, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\2.0.5\AppHandling\Run-AlCops.ps1: Zeile 112
bei <ScriptBlock>, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\2.0.5\AppHandling\Run-AlValidation.ps1: Zeile 495
bei <ScriptBlock>, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\2.0.5\AppHandling\Run-AlValidation.ps1: Zeile 478
bei <ScriptBlock>, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\2.0.5\AppHandling\Run-AlValidation.ps1: Zeile 393
bei <ScriptBlock>, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\2.0.5\AppHandling\Run-AlValidation.ps1: Zeile 367
bei Run-AlValidation, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\2.0.5\AppHandling\Run-AlValidation.ps1: Zeile 365
bei <ScriptBlock>, <Keine Datei>: Zeile 1
```

Will it be possible to work the validation with preprocessors too because the validation is a very good and easy way to check the app!?

Regards
Andree

**Comment by atoader** [comment]
Hi @GeekEding ! Thank you for reporting this issue. We are looking into how we can address it best. Have you encountered any issues while deploying the extension in SaaS from VSCode?

**Comment by GeekEding** [comment]
Hi @atoader !
Actually I don't publish from VSCode to SaaS directly. 
I published it to my SaaS Container from VSCode several times and this worked perfectly.

**Comment by freddydk** [comment]
Have you submitted to AppSource Validation?

**Comment by GeekEding** [comment]
Hi @freddydk ,
Not yet! We have to do some more tests.

**Comment by freddydk** [comment]
Ok, thanks

**Comment by freddydk** [comment]
Shipped in BcContainerHelper 2.0.6
